x-pack/plugins/lens/public/xy_visualization_plugin/xy_expression.tsx

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {
+  Position,
+  Chart,
+  Settings,
+  Axis,
+  LineSeries,
+  getAxisId,
+  getSpecId,
+  AreaSeries,
+  BarSeries,
+} from '@elastic/charts';
+import { Datatable } from '../../../canvas/canvas_plugin_src/functions/types';
+
+/**
+ * This file contains TypeScript type definitions and their equivalent expression
+ * definitions, for configuring and rendering an XY chart. The XY chart serves
+ * triple duty as a bar, line, or area chart.
+ *
+ * The xy_chart expression function serves mostly as a passthrough to the xy_chart_renderer
+ * which does the heavy-lifting.
+ */
+
+/**
+ * Configuration for the chart's legend.
+ */
+export interface LegendConfig {
+  isVisible: boolean;
+  position: Position;
+}
+
+/**
+ * LegendConfig as an expression type.
+ */
+export const legendConfig = {
+  name: 'legendConfig',
+  aliases: [],
+  type: 'legendConfig',
+  help: `Configure a chart's legend`,
+  context: {
+    types: ['null'],
+  },
+  args: {
+    isVisible: { types: ['boolean'] },
+    position: { types: ['string'] },
+  },
+  fn: function fn(_context: any, args: LegendConfig) {
+    return {
+      type: 'legendConfig',
+      ...args,
+    };
+  },
+};
+
+/**
+ * Properties common to all axes.
+ */
+interface AxisConfig {
+  title: string;
+  showGridlines: boolean;
+  position: Position;
+}
+
+/**
+ * AxisConfig expression definition.
+ */
+const axisConfig = {
+  title: { types: ['string'] },
+  showGridlines: { types: ['boolean'] },
+  position: { types: ['string'] },
+};
+
+/**
+ * Y-axis configuration.
+ */
+export interface YConfig extends AxisConfig {
+  accessors: string[];
+}
+
+/**
+ * YConfig as an expression type.
+ */
+export const yConfig = {
+  name: 'yConfig',
+  aliases: [],
+  type: 'yConfig',
+  help: `Configure a chart's y axis`,
+  context: {
+    types: ['null'],
+  },
+  args: {
+    ...axisConfig,
+    accessors: {
+      types: ['string'],
+      multi: true,
+    },
+  },
+  fn: function fn(_context: any, args: YConfig) {
+    return {
+      type: 'yConfig',
+      ...args,
+    };
+  },
+};
+
+/**
+ * X-axis configuration.
+ */
+export interface XConfig extends AxisConfig {
+  accessor: string;
+}
+
+/**
+ * XConfig as an expression type.
+ */
+export const xConfig = {
+  name: 'xConfig',
+  aliases: [],
+  type: 'xConfig',
+  help: `Configure a chart's x axis`,
+  context: {
+    types: ['null'],
+  },
+  args: {
+    ...axisConfig,
+    accessor: { types: ['string'] },
+  },
+  fn: function fn(_context: any, args: XConfig) {
```

Comment:
`any` types shouldn't work because of the updated eslint rule - maybe that's fixed automatically with the enhanced types from Canvas. If not, we can just use `unknown` here, it's unused anyway